Realm/RLMObjectBase.mm

```diff
@@ -330,16 +332,14 @@ void RLMObjectBaseSetObjectSchema(__unsafe_unretained RLMObjectBase *object, __u
 
     Table *table = schema.table;
     if (!table) {
-        return @[];
+        return [RLMEmptyResults emptyResultsWithObjectClassName:schema.className realm:object->_realm];
     }
 
-    size_t col = prop.column;
-    NSUInteger count = object->_row.get_backlink_count(*table, col);
-    NSMutableArray *links = [NSMutableArray arrayWithCapacity:count];
-    for (NSUInteger i = 0; i < count; i++) {
-        [links addObject:RLMCreateObjectAccessor(object->_realm, schema, object->_row.get_backlink(*table, col, i))];
-    }
-    return [links copy];
+    TableView tableView = object->_row.get_table()->get_backlink_view(object->_row.get_index(), table, prop.column);
+    TableView tv(tableView);
+    const Table& constTable = tv.get_parent();
+    auto query = std::make_unique<realm::Query>(constTable, &tv);
```

Comment:
This query will only remain valid until you return from the function, as it just holds a reference to the stack-allocated `tv`. Query only takes ownership of the TableView if you use the overload that takes a unique_ptr.

There's also no real need to create the query here;  `-[RLMResults cloneQuery]` could instead handle the case where _backingQuery is null but _backingView is valid.